docs/markdown/DevGuide_BuildProcessor.md

```diff
@@ -0,0 +1,115 @@
+```
+
+#### Dispatching data to the CPU:
+Inviwo has different data representations, depending on where the data is stored (Hard drive, RAM, GPU RAM) and by which API it is used (e.g. OpenGL, OpenCL). You can request each of those representations using `getRepresentation<RepresentationType>()`. For example you could get a volume representation in RAM as follows:
+```
+auto volume_ram_repr = volumeInport.getData()->getRepresentation<VolumeRAM>();
+```
+However, you cannot directly use this representation to modify your data. This representation basically holds a `void*` pointer to the raw data. In order to safely cast it to a concrete type (`int`, `float`, `vec3`), so that you can use the data, you have to use `dispatch<>()`.
+
+The `dispatch<>()` method takes a lambda as parameter and the lambda is passed the correctly typed data as parameter. That means you can only access your data directly inside `dispatch<>()`. Furthermore, `dispatch<ReturnType>()` has a type parameter as well, which is the return type of the lambda. See the following example:
+```
+volume_ram_repr->dispatch<  std::shared_ptr<Volume>  >([] (auto ram) {
+    using ValueType = util::PrecisionValueType<decltype(ram)>;
+    const size3_t dims = ram->getDimensions();
+
+    dstRam = std::make_shared<VolumeRAMPrecision< typename ValueType> >(dims);
+    const ValueType* srcVol = ram->getDataTyped(); // This is always const for data coming from inports!
+    ValueType* dstData = dstRam->getDataTyped();
+
+    for (size_t i = 0; i < (dims.x * dims.y * dims.z); ++i) {
+        dstData[i] = srcData[i] * ValueType(2);
+    }
+    return std::make_shared<Volume>(dstRam);
+});
+```
+This piece of code takes the volume from the inport (more specifically its `VolumeRAM` representation) and outputs another `Volume` with all values doubled. There are a few things to note here:
+- This works for all possible data types of the volume, it doesn't matter whether it's `float`, `int` or anything else. `ValueType` will be the corresponding type inside `dispatch()`. In order to support all possible types, you have to define the lambda argument with `auto` as is done in the first line. Using this combination of `auto` and the `ValueType` will have your compiler produce `dispatch()` functions for every supported type. Make sure to wrap constants like the `2` inside the `for`-loop in a `ValueType` cast to ensure that its type matches your data's type.
+- The template parameter of `dispatch<>()`, in this case `std::shared_ptr<Volume>`, is the return type of the lambda and `dispatch<>()`. This must be the same for all data types.
+- All the data that comes from an inport, like the `srcVol` in the example, is `const`. If you want to modify the volume, you will have to create a new one as demonstrated in the example.
+- If you need any other variables from outside the scope of the `dispatch<>()`'s lambda, you can pass them into the lambda like so: `dispatch<...>([&my_local_var] (auto ram) { ... })`
+
```

Comment:
Perhaps we can add "Accessing data in OpenGL" and add the corresponding example?